### PR TITLE
Add libwav to kos-ports

### PR DESCRIPTION
**Pull request by andressbarajas**

### CODE REVIEW

File: libwav/files/sndwav.c [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+    /* Used only in reading wav data from a buffer 
+       and not a file */
+    uint32_t buf_offset;  
+   
+} snddrv_hnd;
+
+static snddrv_hnd streams[SND_STREAM_MAX];
+static volatile int sndwav_status = SNDDRV_STATUS_NULL;
+
+static void *sndwav_thread();
```
Comment (gyrovorbis) [inline libwav/files/sndwav.c:74]:
```suggestion
static void *sndwav_thread(void);
```

File: libwav/files/sndwav.c [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+
+void wav_add_filter(wav_stream_hnd_t hnd, wav_filter filter, void *obj) {
+    snd_stream_filter_add(streams[hnd].shnd, filter, obj);
+}
+
+void wav_remove_filter(wav_stream_hnd_t hnd, wav_filter filter, void *obj) {
+    snd_stream_filter_remove(streams[hnd].shnd, filter, obj);
+}
+
+static void *sndwav_thread() {
```
Comment (gyrovorbis) [inline libwav/files/sndwav.c:281]:
```suggestion
static void *sndwav_thread(void) {
```

**Review by gyrovorbis — COMMENTED** [review]
Damn purdy API!

File: libwav/files/sndwav.c [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+/* Keep track of things from the Decoder side */
+#define SNDDEC_STATUS_NULL         0x00
+#define SNDDEC_STATUS_READY        0x01
+#define SNDDEC_STATUS_STREAMING    0x02
+#define SNDDEC_STATUS_PAUSING      0x03
+#define SNDDEC_STATUS_STOPPING     0x04
+#define SNDDEC_STATUS_RESUMING     0x05
+
+#define STREAM_BUFFER_SIZE 65536
```
Comment (DC-SWAT) [inline libwav/files/sndwav.c:26]: Not used

Comment (andressbarajas) [reply libwav/files/sndwav.c:26]: Youre right.  When I updated to dynamic memory I ended up using SND_STREAM_BUFFER_MAX instead which has the same value.

File: libwav/files/sndwav.c [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+
+    if(filename == NULL)
+        return SND_STREAM_INVALID;
+
+    file = fs_open(filename, O_RDONLY);
+
+    if(file == FILEHND_INVALID)
+        return SND_STREAM_INVALID;
+
+    index = snd_stream_alloc(wav_file_callback, SND_STREAM_BUFFER_MAX); // SND_STREAM_BUFFER_MAX/4
```
Comment (DC-SWAT) [inline libwav/files/sndwav.c:142]: This comment a bit confused:
"// SND_STREAM_BUFFER_MAX/4"

Comment (andressbarajas) [reply libwav/files/sndwav.c:142]: It's a stub holder. I guess I can remove it.

**Review by ljsebald — APPROVED** [review]

**Pull request merged**